povertymapping/feature_engineering.py

```diff
@@ -0,0 +1,137 @@
+from typing import Union, Any
+from sklearn.preprocessing import StandardScaler
+import pandas as pd
+
+
+def generate_features(
+    aoi: pd.DataFrame,
+    country_osm: str,
+    ookla_year: int,
+    nightlights_year: int,
+    inplace: bool = False,
+    data_dir: str = settings.DATA_DIR,
+    fill_na: bool = True,
+    fill_na_value: int = 0,
+    scale: bool = True,
+    sklearn_scaler: Any = StandardScaler,
+    scaled_only: bool = False,
+    features_only: bool = False,
+) -> pd.DataFrame:
+    """Generates the base features for an AOI based on
+    DHS, OSM, Ookla, and VIIRS (nighttime lights) data
+
+    Args:
+        aoi (pd.DataFrame): The input AOI dataframe.
+        country_osm (str): The name of country to load features from OSM. For more information about
+            supported countries, refer to Geofabrik documentation: https://download.geofabrik.de/index.html
+        ookla_year (int): The year of Ookla data to use. Earliest available is 2019.
+        nightlights_year (int): The year of nighttime lights data to use. Earliest available is 2012.
+        inplace (bool, optional): Whether to overwrite the original dataframe or not. Defaults to False.
+        data_dir (str, optional): The path to the data directory. Defaults to settings.DATA_DIR.
+        fill_na (bool, optional): Whether to fill missing values with fill_na_value or not. Defaults to True.
+        fill_na_value (int, optional): The value to fill missing values. Defaults to 0.
+        scale_features (bool, optional): Whether to scale the generated features or not. Defaults to True.
+        sklearn_scaler (Any, optional): The scikit-learn scaler to use. Defaults to StandardScaler.
+        scaled_only (bool, optional): Whether to return only the scaled features or not. Defaults to False.
+        features_only (bool, optional): Whether to return only the generated features or not. Defaults to False.
+
+    Returns:
+        aoi (pd.DataFrame): The AOI dataframe with its new features.
+    """
+    # Make a copy of the dataframe to avoid overwriting the input data
+    if not inplace:
+        aoi = aoi.copy()
+
+    # Store list of input columns
+    input_cols = aoi.columns
+
+    # Instantiate data managers for Ookla and OSM
+    # This auto-caches requested data in RAM, so next fetches of the data are faster.
+    osm_data_manager = OsmDataManager(cache_dir=data_dir / "data_cache")
+    ookla_data_manager = OoklaDataManager(cache_dir=data_dir / "data_cache")
+
+    # Add in OSM features
+    aoi = osm.add_osm_poi_features(aoi, country_osm, osm_data_manager)
+    aoi = osm.add_osm_road_features(aoi, country_osm, osm_data_manager)
+
+    # Add in Ookla features
+    aoi = ookla.add_ookla_features(aoi, "fixed", ookla_year, ookla_data_manager)
+    aoi = ookla.add_ookla_features(aoi, "mobile", ookla_year, ookla_data_manager)
+
+    # Add in the nighttime lights features
+    aoi = nightlights.generate_nightlights_feature(aoi, nightlights_year)
```

Comment:
+1 to all the cache_dir updates to make it consistent to use `~/.geowrangler` already. Just noting you might have to move your cache files @tm-jc-nacpil  because of this.